conformance/utils/http/http.go

```diff
@@ -38,9 +38,13 @@ type ExpectedResponse struct {
 	// expected to match Request.
 	ExpectedRequest *ExpectedRequest
 
-	StatusCode int
-	Backend    string
-	Namespace  string
+	// TODO: maybe move these into a dedicated type named Response?
```

Comment:
We should either:

a) resolve/remove this TODO as a part of this PR
b) if there's sufficient cause to follow up on this after this PR, create an issue for it and link it here so the TODO doesn't get lost.

Reply:
I think there is be sufficient cause, but I don't think it'd be wise to mix this with the current PR changes. I'll create an issue for this and open a follow up PR after this gets merged.